proposal.rst

```diff
@@ -239,7 +301,82 @@ Project Timeline
 .. any university classes or exams, vacations, etc), make sure you include them
 .. in your project timeline.
 
-FILLME
+MONTH 1  : 
+
+BIG TASK NAME : DOCUMENTATION STEP
+
+Week 1 
+	- Read and understand code-base further. 
+	- Create a new VagrantFile and make sure that it installs all the xapian dependencies. I have a doc where I have all commands which helped me with my Xapian setup.
+	- Update the "Installation" page with vagrant walk through tutorial.
+
+Week 2 
+	- Re-work on "Practical Example" tutorial to make it more user-friendly. 
+	- Add Omega documentation (in Xapian docs) for indexing, and how to obtain a front-end html(omindex, omega "query" [arguments])
```

Comment:
While I agree that Omega's documentation is very limited, GSoC is (sadly) expressly not about writing documentation. You can write documentation as part of delivering something else; writing example code that exercises Xapian or Omega would I think just about count. Given it's possible to get Omega running without writing new code, I'm wondering what the HTML frontend you'd implement and then document here would be, and how it'd be different to what's existing. (Again, the existing templates aren't great — as you've spotted, they could do with being moved to html5 at the least.)

Reply:
So first part will be little code examples of omindex and /bin/omega from https://xapian.org/docs/omega/quickstart.html. Also I followed this and received a html file(output of /usr/lib/omega/bin/omega 'P=my search terms' HITSPERPAGE=10 ), but the clicking search goes to page where it says "file not found". I do not have perfect understanding of cgi templates, yet however I would like to make that work and include in Xapian docs little intro to Omega. Also some coding examples which connects omega and xapian-delve(if I create db through xaipan-delve and omega reads from that database.)

So my steps would be : Get templates workable with cgi as suggested here : https://xapian.org/docs/omega/cgiparams.html, and then think about upgrading to more modern HTML5.

Reply:
We'd definitely expect you to get on top of CGI quite quickly — it's not complex and there are plenty of documents discussing it. (For instance, for Apache: https://httpd.apache.org/docs/2.4/howto/cgi.html — and indeed we have a worked example on our wiki, although it's for Xapian 1.0 so perhaps should be updated. I'd recommend working through that example and updating it if needed; it will give you a much better idea of what you might need to do for Omega documentation.)

The problem you're encountering, as I believe was pointed out on IRC (but you may have missed) is that you're trying to treat the HTML generated by a test run of omega as if it's a live site. If you set up omega to run as  CGI with a web server, you shouldn't have this problem.